### ISSUE TITLE
Create classification fails - merging autopopulate literature

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#871]
View details in Rollbar: [https://app.rollbar.com/a/jimmy.andrews/fix/item/VariantGrid/4517](https://app.rollbar.com/a/jimmy.andrews/fix/item/VariantGrid/4517)

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.10/dist-packages/django/views/decorators/http.py", line 43, in inner
    return func(request, *args, **kwargs)
  File "/opt/variantgrid/classification/views/views.py", line 232, in create_classification
    return redirect(create_classification_object(request).get_edit_url())
  File "/opt/variantgrid/classification/views/views.py", line 260, in create_classification_object
    classification = create_classification_for_sample_and_variant_objects(request.user, sample,
  File "/opt/variantgrid/classification/autopopulate_evidence_keys/autopopulate_evidence_keys.py", line 44, in create_classification_for_sample_and_variant_objects
    classification_auto_populate_fields(classification, genome_build,
  File "/opt/variantgrid/classification/autopopulate_evidence_keys/autopopulate_evidence_keys.py", line 118, in classification_auto_populate_fields
    return classification.patch_value(auto_data.data,
  File "/opt/variantgrid/classification/autopopulate_evidence_keys/evidence_from_variant.py", line 80, in data
    for key, value in other_data.data.items():
  File "/opt/variantgrid/classification/autopopulate_evidence_keys/evidence_from_variant.py", line 82, in data
    all_data[key] = '\n'.join([all_data[key], value])
TypeError: sequence item 1: expected str instance, dict found
```

**Comment by davmlaw** [comment]
```
all_data[key] = '\n'.join([all_data[key], value])
```

Dies as value = ```{'value': None} ```

**Comment by davmlaw** [comment]
@TheMadBug if you want to take a [look at my fix](https://github.com/SACGF/variantgrid/commit/ff83960216606466c981bed9433604aecaebbb39) and make sure it's good

I swear we had a method to pull out values (that can be dict with key of 'value') but I can't find it.

Though if you search for ```isinstance.*Mapping``` (with regex on) in PyCharm you can see a heap of code that is basically doing this from what I can tell

**Comment by TheMadBug** [comment]
(from Slack before I saw this GitHub comment)

There are several sources to auto populate data with - annotations being one, and other classifications being another.
There’s a bunch of caching that goes on (before I knew about @cached_property and invalidate - this is close to the first code I wrote for VG), so _all_data is a cache of the data that gets cleared when something would invalidate that (like merging another AutoPopulate).
So likely 99% of the time the autopopulate source was just annotations that was just a simple dict, but in that case it was a dict and another classification so broke.
Now the problem with AUTOPOPULATE_MERGE_KEYS of literature, it means its just going to append the literature together. So we might get a lot of duplication.
But improving that can always be another task - making it not crash is obviously higher priority.
